tools/roxvet/analyzers/uncheckedifassign/analyzer.go

```diff
@@ -24,18 +20,21 @@ var Analyzer = &analysis.Analyzer{
 }
 
 func getAssignedObjects(ifStmt *ast.IfStmt) map[*ast.Object]struct{} {
-	initAssign, _ := ifStmt.Init.(*ast.AssignStmt)
-	if initAssign == nil {
-		return nil
-	}
-
 	objs := make(map[*ast.Object]struct{})
-	for _, expr := range initAssign.Lhs {
-		idExpr, _ := expr.(*ast.Ident)
-		if idExpr == nil || idExpr.Obj == nil || idExpr.Obj.Kind != ast.Var {
-			continue
+	switch initAssign := ifStmt.Init.(type) {
+	case *ast.AssignStmt:
+		for _, expr := range initAssign.Lhs {
+			idExpr, _ := expr.(*ast.Ident)
+			if idExpr == nil || idExpr.Obj == nil || idExpr.Obj.Kind != ast.Var {
+				continue
+			}
+			objs[idExpr.Obj] = struct{}{}
+		}
+	case *ast.IncDecStmt:
```

Comment:
is this `x += 1` or `x++`?

Reply:
This is `x++`. `x += 1` is actually an `*ast.AssignStmt`

Reply:
I don't think I've ever seen this used in an `if`. Do we have any examples?

Reply:
Yeah, for example `pkg/networkgraph/tree/radix.go:160`. I saw this failing on `make roxvet` with the earlier changes.